services/vectaraSearcherPreprocessorSvc.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+angular.module('o19s.splainer-search')
+    .service('vectaraSearcherPreprocessorSvc', [
+        'queryTemplateSvc',
+        'defaultVectaraConfig',
+        function vectaraSearcherPreprocessorSvc(queryTemplateSvc, defaultVectaraConfig) {
+            const self = this;
+
+            // Functions
+            self.prepare  = prepare;
+
+            const replaceQuery = function(args, queryText) {
+                // Allows full override of query if a JSON friendly format is sent in
```

Comment:
is this because there is more than one format?

Reply:
This is just a leftover from ES code. I don't understand under what circumstances `queryText` is an `Object` and when it would be a `String`, so I just left that in here :/

Reply:
Can we remove that then?   Just to have less cognitive load?